src/Microsoft.DocAsCode.Build.OverwriteDocuments/OverwriteUtility.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.Build.OverwriteDocuments
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text.RegularExpressions;
+
+    public class OverwriteUtility
+    {
+        private static readonly Regex OPathRegex =
+            new Regex(
+                @"^(?<propertyName>[\w\.\-]+)(\[(?<key>[\w\.\-]+)=""(?<value>[\w\(\)\.\{\}\[\]\|\/@`<>: ]+)""\])?/?",
```

Comment:
>\w\.\- [](start = 36, length = 6)

Although the property name can support any chars in YAML spec, but as we borrow the syntax from XPath, we want to limit the name as a valid **XML** name: [https://www.w3.org/TR/2008/REC-xml-20081126/#NT-NameStartChar](https://www.w3.org/TR/2008/REC-xml-20081126/#NT-NameStartChar)